test/framework/k8s_controlplane.go

```diff
@@ -227,10 +227,19 @@ func (c *K8sControlPlane) GetKDSServerAddress() string {
 	// As EKS and AWS generally returns dns records of load balancers instead of
 	//  IP addresses, accessing this data (hostname) was only tested there,
 	//  so the env var was created for that purpose
-	if UseLoadBalancer() && IsInEKS() {
+	if UseLoadBalancer() {
 		svc := c.GetKumaCPSvcs()[0]
 
-		return "grpcs://" + svc.Status.LoadBalancer.Ingress[0].Hostname + ":" + strconv.FormatUint(loadBalancerKdsPort, 10)
+		ip, hostname := svc.Status.LoadBalancer.Ingress[0].IP, svc.Status.LoadBalancer.Ingress[0].Hostname
```

Comment:
I'm a bit hesitated that we can retrieve a service that has still pending external IP.
I think it would be more safe to wait here until we have either IP or hostname.

We can go with it and see if it works, but this is potentially something flaky

Reply:
We decided to treat it as potential technical debt and observe if the tests are being flake because of it, and if so, followup with the fixes